### DISCUSSION TITLE
Slow development process especially for React Components

### DISCUSSION THREAD

**Discussion by cgeorgilakis** [keycloak/keycloak#23893] [Q&A]
You have instructions in [Running in Keycloak development mode](https://github.com/keycloak/keycloak/blob/main/quarkus/README.md#running-in-keycloak-development-mode) for developing Keycloak with Quarkus.
However, after version 22 development process is very slow - especially for React components. For any change we need to build specific folder and after build Quarkus ( from quarkus folder : mvn clean install -DskipTests). And after run/debug quarkus-run.jar. The need to build based on maven make things worse.

I have tried also to run _org.keycloak.quarkus._private.IDELauncher_ from IntelliJ without success. I get the following error : 
`Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: org/wildfly/common/lock/SpinLock [in thread "main"]`

How can we speed up development process - especially using IntelliJ?
How can we overcome IDELauncher failure? If this helps us developing faster.

**Comment by ssilvert** [comment]
Are you developing React components on the account or admin consoles?  If so, you don't need to do a build to see your changes.  See the readme for admin console.

https://github.com/keycloak/keycloak/tree/main/js/apps/admin-ui

**Reply by cgeorgilakis** [reply]
Thank you. This works fine 😃. It will be good to have a link to this documentation in the [Keycloak on Quarkus documentation](https://github.com/keycloak/keycloak/blob/main/quarkus/README.md).

For Java components with Intellij can somehow autobuild work? I run/debug Keycloak with quarkus-run.jar.
